packages/client/src/provider/in-memory-provider/in-memory-provider.ts

```diff
@@ -59,16 +59,11 @@ export class InMemoryProvider implements Provider {
       .filter(([key, value]) => this._flagConfiguration[key] !== value)
       .map(([key]) => key);
 
-    this.status = ProviderStatus.STALE;
-    this.events.emit(ProviderEvents.Stale);
-
     this._flagConfiguration = { ...flagConfiguration };
-    this.events.emit(ProviderEvents.ConfigurationChanged, { flagsChanged });
 
     try {
       await this.initialize(this._context);
-      // we need to emit our own events in this case, since it's not part of the init flow.
-      this.events.emit(ProviderEvents.Ready);
```

Comment:
Why is this no longer relevant?

Reply:
See my [other comment](https://github.com/open-feature/js-sdk/pull/796#discussion_r1471597115).

There's no need for STALE/READY or anything here. This is a method completely unique to the in-memory provider that changes it's config. It should simply emit a config change event.